caikit/core/data_model/dataobject.py

```diff
@@ -215,9 +235,11 @@ class _DataobjectConverter(DataclassConverter):
     def get_concrete_type(self, entry: Any) -> Any:
         """Also include data model classes and enums as concrete types"""
         unwrapped = self._resolve_wrapped_type(entry)
-        if (isinstance(unwrapped, type) and issubclass(unwrapped, DataBase)) or hasattr(
-            unwrapped, "_proto_enum"
-        ):
+        if (
```

Comment:
This was interesting. Now that the class being decorated is _already_ a derived `DataBase` class, this was triggering to detect it as an already-converted "concrete type reference." The check on `unwrapped._proto_class is not None` eliminates this case and indicates that this class has yet to be converted.

Reply:
Any chance that context can be thrown into a comment?